apps/cli/src/index.tsx

```diff
@@ -1,45 +1,253 @@
+    "config" in valueAsObject &&
+    typeof valueAsObject.config === "object" &&
+    "allow_respond" in valueAsObject.config &&
+    "allow_accept" in valueAsObject.config &&
+    "allow_edit" in valueAsObject.config &&
+    "allow_ignore" in valueAsObject.config
+  );
+}
+
 const App: React.FC = () => {
   const [authPrompt, setAuthPrompt] = useState<null | boolean>(null);
   const [authInput, setAuthInput] = useState("");
-  const [message, setMessage] = useState<string | null>(null);
   const [exit, setExit] = useState(false);
   const [authStarted, setAuthStarted] = useState(false);
   const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [repos, setRepos] = useState<any[]>([]);
+  const [selectedRepo, setSelectedRepo] = useState<any | null>(null);
+  const [selectingRepo, setSelectingRepo] = useState(false);
+  const [waitingForInstall, setWaitingForInstall] = useState(false);
+  const [installChecked, setInstallChecked] = useState(false);
+  const [installError, setInstallError] = useState<string | null>(null);
+  const [appSlug, setAppSlug] = useState(process.env.GITHUB_APP_NAME || "");
+  const INSTALLATION_CALLBACK_URL = process.env.GITHUB_CALLBACK_URL || "";
+  const [pollingForToken, setPollingForToken] = useState(false);
+  const [logs, setLogs] = useState<string[]>([]);
+  const [prompt, setPrompt] = useState<string>("");
+  const [plannerFeedback, setPlannerFeedback] = useState<string | null>(null);
+  const [streamingPhase, setStreamingPhase] = useState<
+    "streaming" | "awaitingFeedback" | "done"
+  >("streaming");
+  const [threadId, setThreadId] = useState<string | null>(null);
+  const [plannerThreadId, setPlannerThreadId] = useState<string | null>(null);
+  const [hasStartedChat, setHasStartedChat] = useState(false);
+  const [client, setClient] = useState<Client | null>(null);
+  const [loadingRepos, setLoadingRepos] = useState(false);
+  const [loadingLogs, setLoadingLogs] = useState(false);
+
+  const sendInterruptMessage = useCallback(
+    async (message: string) => {
+      if (!client || !threadId || !selectedRepo) {
+        return;
+      }
+
+      setLogs((prev) => [...prev, `📤 Interrupt: "${message}"`]);
+
+      try {
+        const [owner, repoName] = selectedRepo.full_name.split("/");
+        const interruptInput = {
+          messages: [
+            {
+              id: uuidv4(),
+              type: "human",
+              content: [{ type: "text", text: message }],
+            },
+          ],
+          targetRepository: {
+            owner,
+            repo: repoName,
+            branch: selectedRepo.default_branch || "main",
+          },
+        };
+        await client.runs.create(threadId, MANAGER_GRAPH_ID, {
+          input: interruptInput,
+          config: { recursion_limit: 400 },
+          ifNotExists: "create",
+          streamResumable: true,
+          multitaskStrategy: "enqueue",
+          streamMode: ["values", "messages"],
```

Comment:
the `streamMode` arr should prob be exported as a constant from `shared` so that we can reuse here, and everywhere else in the app to ensure consistent stream modes
```suggestion
          streamMode: ["values", "updates", "messages"],
```